### ISSUE TITLE
SwitchProducer does not meet the expectations of the mass replacement tools

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#28852]
`SwitchProducer` does not meet the expectations of the mass replacement tools in `FWCore.ParameterSet.MassReplace`. Noticed in https://github.com/cms-sw/cmssw/issues/28850#issuecomment-581149286
```
  File "/data/cmsbld/jenkins/workspace/auto-builds/CMSSW_11_1_0_pre2_Patatrack-slc7_amd64_gcc820/build/CMSSW_11_1_0_pre2_Patatrack-build/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_1_0_pre2_Patatrack/python/FWCore/ParameterSet/MassReplace.py", line 69, in enter
    self.doIt(visitee, label)
  File "/data/cmsbld/jenkins/workspace/auto-builds/CMSSW_11_1_0_pre2_Patatrack-slc7_amd64_gcc820/build/CMSSW_11_1_0_pre2_Patatrack-build/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_1_0_pre2_Patatrack/python/FWCore/ParameterSet/MassReplace.py", line 20, in doIt
    type = value.pythonTypeName()
AttributeError: 'EDProducer' object has no attribute 'pythonTypeName'
```

**Comment by makortel** [comment]
assign core

**Comment by makortel** [comment]
Fixed in #28895
